dotCMS/src/main/java/com/dotmarketing/servlets/ajax/Trashme.java

```diff
@@ -0,0 +1,245 @@
+        fieldsSearch = (Map<String, String>) lastSearch.get("fieldsSearch");
+      }
+      if (lastSearch.get("categories") != null) {
+        categories = (List<String>) lastSearch.get("categories");
+      }
+      if (UtilMethods.isSet(lastSearch.get("showDeleted"))) {
+        showDeleted = (Boolean) lastSearch.get("showDeleted");
+      }
+      if (UtilMethods.isSet(lastSearch.get("filterSystemHost"))) {
+        filterSystemHost = (Boolean) lastSearch.get("filterSystemHost");
+      }
+      if (UtilMethods.isSet(lastSearch.get("filterLocked"))) {
+        filterLocked = (Boolean) lastSearch.get("filterLocked");
+      }
+      if (lastSearch.get("filterUnpublish") != null)
+        filterUnpublish = (Boolean) lastSearch.get("filterUnpublish");
+      if (UtilMethods.isSet(lastSearch.get("page"))) {
+        currpage = (Integer) lastSearch.get("page");
+      }
+      if (UtilMethods.isSet(lastSearch.get("orderBy"))) {
+        orderBy = (String) lastSearch.get("orderBy");
+      }
+
+    }
+
+    if (!UtilMethods.isSet(tryTypeId)) {
+      if (session.getAttribute("selectedStructure") != null) {
+        String longSelectedStructure = (String) session.getAttribute("selectedStructure");
+        if (UtilMethods.isSet(longSelectedStructure)) {
+          tryTypeId = longSelectedStructure.toString();
+        }
+      }
+    }
+
+    if (!UtilMethods.isSet(tryTypeId)
+        || !contentTypes.contains(contentTypeAPI.find(tryTypeId))) {
+
+      tryTypeId = "catchall";
+
+    }
+    ContentType typeSelected=contentTypeAPI.find(tryTypeId);
+    List<Field> fields = new ArrayList<Field>();
+    try {
+      
+      fields = typeSelected.fields();
+    } catch (Exception e) {
+      Logger.debug(this.getClass(), e.getMessage());
+    }
+    boolean hasNoSearcheableHostFolderField = false;
+    boolean hasHostFolderField = false;
+    for (Field field : fields) {
+      if (field instanceof HostFolderField) {
+        if (APILocator.getPermissionAPI().doesUserHavePermission(APILocator.getHostAPI().findSystemHost(), PermissionAPI.PERMISSION_READ,
+            user, true)) {
+          hasHostFolderField = true;
+        }
+        if (!field.searchable()) {
+          hasNoSearcheableHostFolderField = true;
+        }
+        break;
+      }
+    }
+
+    if (fieldsSearch == null || !UtilMethods.isSet(fieldsSearch.get("conHost")) || hasNoSearcheableHostFolderField) {
+      fieldsSearch.put("conHost", (String) session.getAttribute(com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID));
+    }
+
+    String crumbtrailSelectedHostId = (String) session.getAttribute(com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID);
+    if ((crumbtrailSelectedHostId == null) || crumbtrailSelectedHostId.equals("allHosts"))
+      crumbtrailSelectedHostId = "";
+
+    String structureInodesList = "";
+    String structureVelocityVarNames = "";
+
+    for (ContentType type: contentTypes) {
+
+      if (structureInodesList != "") {

```

Comment:
this can be replaced by: 
```
     if (!structureInodesList.equals("")) {
        structureInodesList += ";";
      } 
      structureInodesList += type.id();
```

Reply:
class deleted